### PR TITLE
startable without audio

### PR DESCRIPTION
**Pull request by lyrra**
Enables testing/working without any audio.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [ ] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [ ] I made sure the code compiles on my machine
- [ ] I made sure there are no unnecessary changes in the code
- [ ] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [ ] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [ ] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

File: src/framework/audio/internal/platform/lin/linuxaudiodriver.cpp [hunk 1]
```diff
@@ -88,8 +88,10 @@ static void alsaCleanup()
     if (_alsaData->threadHandle) {
         pthread_join(_alsaData->threadHandle, nullptr);
     }
-    snd_pcm_drain(_alsaData->alsaDeviceHandle);
-    snd_pcm_close(_alsaData->alsaDeviceHandle);
+    if (_alsaData->alsaDeviceHandle) {
```
Comment (vpereverzev) [inline src/framework/audio/internal/platform/lin/linuxaudiodriver.cpp:91]: how is that related to tests w/o the "audio" module? Linux audio driver is a part of it

File: src/playback/view/mixerpanelmodel.cpp [hunk 2]
```diff
@@ -40,22 +40,24 @@ void MixerPanelModel::load()
...
-    });
-
-    playback()->tracks()->trackAdded().onReceive(this, [this](const TrackSequenceId sequenceId, const TrackId trackId) {
-        addItem(sequenceId, trackId);
-    });
-
-    playback()->tracks()->trackRemoved().onReceive(this, [this](const TrackSequenceId /*sequenceId*/, const TrackId trackId) {
-        removeItem(trackId);
-    });
+    if (playback()->tracks()) {
```
Comment (vpereverzev) [inline src/playback/view/mixerpanelmodel.cpp:43]: just put

if (playback()->tracks()) {
    return;
}

at the beginning

**Comment by vpereverzev** [comment]
If you're interested in testing of app w/o some module, you might be interested in using "stub" classes

File: src/playback/view/mixerpanelmodel.cpp (hunk 2 above)
Comment (lyrra) [reply src/playback/view/mixerpanelmodel.cpp:43]: I've pushed your suggestion.

File: src/framework/audio/internal/platform/lin/linuxaudiodriver.cpp (hunk 1 above)
Comment (lyrra) [reply src/framework/audio/internal/platform/lin/linuxaudiodriver.cpp:91]: I guess not. If so perhaps this is a quick-fix and the proper fix is to not load the linux-audio-module.
This patch addresses those users who either has a broken linux-audio stack, or purposefully run musescore without any loaded audiodriver.

File: src/playback/internal/playbackcontroller.cpp [hunk 3]
```diff
@@ -145,6 +149,10 @@ void PlaybackController::seek(const midi::tick_t tick)
         return;
     }
 
+    if (! playback()->player()) {
```
Comment (Jojo-Schmitz) [inline src/playback/internal/playbackcontroller.cpp:152]: codestyle issue

File: src/playback/internal/playbackcontroller.cpp [hunk 4]
```diff
@@ -296,6 +304,9 @@ void PlaybackController::rewind(const ActionData& args)
     IF_ASSERT_FAILED(playback()) {
         return;
     }
+    if (! playback()->player()) {
```
Comment (Jojo-Schmitz) [inline src/playback/internal/playbackcontroller.cpp:307]: same

File: src/playback/internal/playbackcontroller.cpp [hunk 5]
```diff
@@ -336,6 +347,10 @@ void PlaybackController::resume()
         return;
     }
 
+    if (! playback()->player()) {
```
Comment (Jojo-Schmitz) [inline src/playback/internal/playbackcontroller.cpp:350]: same

File: src/playback/internal/playbackcontroller.cpp [hunk 6]
```diff
@@ -450,6 +465,10 @@ void PlaybackController::hideLoop()
...
```
Comment (Jojo-Schmitz) [inline src/playback/internal/playbackcontroller.cpp:468]: same

File: src/playback/view/mixerpanelmodel.cpp [hunk 7]
```diff
@@ -40,6 +40,9 @@ void MixerPanelModel::load()
 {
     TrackSequenceId sequenceId = controller()->currentTrackSequenceId();
 
+    if (! playback()->tracks()) {
```
Comment (Jojo-Schmitz) [inline src/playback/view/mixerpanelmodel.cpp:43]: same

Comment (lyrra) [reply src/playback/view/mixerpanelmodel.cpp]: Thanks! pushed a cosmetic fix.

File: src/framework/audio/internal/platform/lin/linuxaudiodriver.cpp (hunk 1 above)
Comment (lyrra) [reply src/framework/audio/internal/platform/lin/linuxaudiodriver.cpp:91]: Further guessing the proper fix should take place in src/framework/audio/audiomodule.cpp, where we unconditionally load alsa if on linux. Until that is done, this patch is a stop-gap solution to avoid crashing.

**Comment by lyrra** [comment]
This patch is relevant until both alsa is made unconditionally loaded. And jack is implemented as a second audio output option.

**Comment by RomanPudashkin** [comment]
The original crash has been fixed, see: https://github.com/musescore/MuseScore/issues/10433#issuecomment-1263330016
